dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/h12/grpc/GrpcCompositeCodecFactory.java

```diff
@@ -22,19 +22,14 @@
 import org.apache.dubbo.remoting.http12.message.HttpMessageDecoderFactory;
 import org.apache.dubbo.remoting.http12.message.HttpMessageEncoderFactory;
 import org.apache.dubbo.remoting.http12.message.MediaType;
-import org.apache.dubbo.remoting.utils.UrlUtils;
 import org.apache.dubbo.rpc.model.FrameworkModel;
 
 @Activate
 public class GrpcCompositeCodecFactory implements HttpMessageEncoderFactory, HttpMessageDecoderFactory {
 
     @Override
     public HttpMessageCodec createCodec(URL url, FrameworkModel frameworkModel, String mediaType) {
-        String serializeName = UrlUtils.serializationOrDefault(url);
-        WrapperHttpMessageCodec wrapperHttpMessageCodec = new WrapperHttpMessageCodec(url, frameworkModel);
-        wrapperHttpMessageCodec.setSerializeType(serializeName);
-        ProtobufHttpMessageCodec protobufHttpMessageCodec = new ProtobufHttpMessageCodec();
-        return new GrpcCompositeCodec(protobufHttpMessageCodec, wrapperHttpMessageCodec);
+        return new GrpcCompositeCodec(url, frameworkModel, mediaType);
```

Comment:
**Security Issue**: The method createCodec directly uses the URL, frameworkModel, and mediaType parameters without validating them, potentially leading to security vulnerabilities if these parameters are user-controlled or can be manipulated. <br> **Fix**: Ensure the URL, frameworkModel, and mediaType parameters are validated against expected values before using them in the createCodec method. <br> **Code Suggestion**: 
 ```
 public HttpMessageCodec createCodec(URL url, FrameworkModel frameworkModel, String mediaType) {
     // Validate URL, frameworkModel, and mediaType parameters
     if (url == null || frameworkModel == null || mediaType == null) {
         throw new IllegalArgumentException("Invalid parameters for codec creation.");
     }
     return new GrpcCompositeCodec(url, frameworkModel, mediaType);
 }
 ```